src/database/handler.py

```diff
@@ -0,0 +1,114 @@
+        # create PostgreSQL engine to connect to the database
+        self.engine = create_engine(
+            f'postgresql://{settings.DATABASE_USERNAME}:{settings.DATABASE_PASSWORD}@{settings.DATABASE_HOST}:{settings.DATABASE_PORT}/{settings.DATABASE_NAME}',
+            pool_pre_ping=True)
+
+        # create the database if does not exist
+        if not database_exists(self.engine.url):
+            create_database(self.engine.url)
+
+        # connect to the database
+        self.engine.connect()
+
+        # create tables if do not exist
+        Base.metadata.create_all(self.engine, checkfirst=True)
+
+        # create a configured "Session" class
+        self.Session = sessionmaker(bind=self.engine)
+
+    def close(self):
+        # TODO: close database connection
+        pass
+
+    def get_document(self, document_id: str) -> Document:
+        # create a Session
+        session = self.Session()
+
+        # get the document from database
+        document = session.query(Document).get(document_id)
+
+        return document
+
+    def insert_document(self,
+                        document_id: str,
+                        document_name: str,
+                        document_status: DocumentStatus = DocumentStatus.IN_QUEUE.value):
+        # create a Session
+        session = self.Session()
+
+        # create database object
+        document = Document(id=document_id, name=document_name, status=document_status)
+
+        # add to the session and commit changes
+        session.add(document)
+        session.commit()
+
+    def update_document(self,
+                        document_id: str,
+                        document_name: str = None,
+                        document_pages_number: int = None,
+                        document_status: DocumentStatus = None):
+        # create a Session
+        session = self.Session()
+
+        # create patch object
+        patch = {}
+
+        if document_name:
+            patch['name'] = document_name
+
+        if document_pages_number:
+            patch['pages_number'] = document_pages_number
+
+        if document_status:
+            patch['status'] = document_status.value
+
+        # update database object and commit changes
+        session.query(Document).filter(Document.id == document_id).update(patch)
+        session.commit()
+
+    def get_page(self, document_id: str, page_id: str) -> Page:
+        # create a Session
+        session = self.Session()
+
+        # get the document from database
+        page = session.query(Page).get((page_id, document_id))
+
+        return page
+
+    def insert_pages(self, document_id: str, pages_data: List[bytes]):
+        # create a Session
+        session = self.Session()
+
+        # create database objects and add them to the session
+        for page_id, page_data in enumerate(pages_data):
+            page = Page(id=page_id, document_id=document_id, data=page_data)
+            session.add(page)
```

Comment:
```suggestion
        session.add_all((Page(id=page_id, document_id=document_id, data=page_data)
                         for page_id, page_data in enumerate(pages_data)))
```
just a phylosophical: how do you like page being numbered 0?

Reply:
:D I am programmer, this is my normal modus operandi... right, I agree, starting pages by 1 should be more user friendly... but as always it depends... if this would be just one microservice somewhere in the middle and the REST API would be used by other microservice, it would worth to keep indexing from zero as the code calling this endpoint would probably use loop over the range of number of pages and so it would be better to keep it as it is.

Reply:
In our system, we have one "inside" microservice that uses the 0-based numbering and one "user-facing" that uses the 1-based numbering. To me, the worst part is the inconsistency and the ever necessary conversion between one another.

Reply:
I agree, I do not like inconsistency as well, to be honest I rather sacrifice some best practices a little bit (to some limits) to be more aligned (especially with mutiple squads across the world). At the end it is always about team performance and bringing the value to the customer.